src/Microsoft.DotNet.ProjectModel/Resolution/FrameworkReferenceResolver.cs

```diff
@@ -207,7 +208,9 @@ private FrameworkInformation GetFrameworkInformation(NuGetFramework targetFramew
         private static FrameworkInformation GetFrameworkInformation(NuGetFramework targetFramework, string referenceAssembliesPath)
         {
             // Check for legacy frameworks
-            if (targetFramework.IsDesktop() && targetFramework.Version <= new Version(3, 5))
+            if (targetFramework.IsDesktop() &&
+                targetFramework.Version <= new Version(3, 5) &&
+                PlatformServices.Default.Runtime.OperatingSystemPlatform == Platform.Windows)
```

Comment:
Move this to the first condition, since it's easiest and excludes all the other conditions. Plus it probably saves us like 0.000001ns :)